include/alpaka/test/Check.hpp

```diff
@@ -14,7 +14,16 @@
         {                                                                                                             \
             if(!(expression))                                                                                         \
             {                                                                                                         \
-                acc.cout << "ALPAKA_CHECK failed because '!(" << #expression << ")'\n";                               \
+                /*if constexpr(alpaka::accMatchesTags<   TODO                                                         \
+                                 decltype(acc),                                                                       \
+                                 alpaka::TagCpuSycl,                                                                  \
+                                 alpaka::TagFpgaSyclIntel,                                                            \
+                                 alpaka::TagFpgaSyclXilinx,                                                           \
+                                 alpaka::TagGpuSyclIntel,                                                             \
+                                 alpaka::TagGenericSycl>)                                                             \
+                    acc.cout << "ALPAKA_CHECK failed because '!(" << #expression << ")'\n";                           \
+                else                                                                                                  \
+                    printf("ALPAKA_CHECK failed because '!(%s)'\n", #expression);*/                                   \
```

Comment:
Please finish the implementation here.

Reply:
I've not found any indication about the `printf` not working on FPGAs, but talking with intel developers the opinion is that even if the `sycl::stream` is not reliable (e.g. sometimes does not print anything), that's the standard. `printf` is more reliable and powerful, but is still experimental in SYCL/oneAPI, so the suggestion is to **not** drop completely the `sycl::stream` for now.